File: dashlane-project/Dashlane/src/main/java/com/dashlane/autofill/api/dagger/AutofillApiComponentModule.java
package com.dashlane.autofill.api.dagger;

import com.dashlane.autofill.changepassword.AutoFillChangePasswordConfiguration;
import com.dashlane.guidedpasswordchange.OnboardingGuidedPasswordChangeActivity;

import dagger.Module;
import dagger.Provides;
import kotlin.Pair;

@Module
public class AutofillApiComponentModule {

    @Provides
    AutoFillChangePasswordConfiguration providesAutoFillChangePasswordConfiguration() {
        Pair<String, String> domainUsername =
                OnboardingGuidedPasswordChangeActivity.Companion.getCurrentDomainUsername();
        if (domainUsername == null) {
            return new AutoFillChangePasswordConfiguration();
        }
        return new AutoFillChangePasswordConfiguration(
                domainUsername.getFirst(),
                domainUsername.getSecond(),
                () -> {
                    
                    OnboardingGuidedPasswordChangeActivity.Companion.setItemUpdated(true);
                    return null;
                },
                () -> {
                    
                    OnboardingGuidedPasswordChangeActivity.Companion.setCurrentDomainUsername(null);
                    return null;
                }
        );
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/autofill/api/dagger/AutofillApiComponentModule.kt
package com.dashlane.autofill.api.dagger

import com.dashlane.autofill.changepassword.AutoFillChangePasswordConfiguration
import com.dashlane.guidedpasswordchange.OnboardingGuidedPasswordChangeActivity
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
object AutofillApiComponentModule {

    @Provides
    fun providesAutoFillChangePasswordConfiguration(): AutoFillChangePasswordConfiguration {
        val domainUsername = OnboardingGuidedPasswordChangeActivity.currentDomainUsername

        return if (domainUsername == null) {
            AutoFillChangePasswordConfiguration()
        } else {
            AutoFillChangePasswordConfiguration(
                filterOnDomain = domainUsername.first,
                filterOnUsername = domainUsername.second,
                onItemUpdated = {
                    OnboardingGuidedPasswordChangeActivity.itemUpdated = true
                },
                onDomainChanged = {
                    OnboardingGuidedPasswordChangeActivity.currentDomainUsername = null
                }
            )
        }
    }
}